pkg/terraform/exec/plugins/vsphereprivate/config.go

```diff
@@ -0,0 +1,67 @@
+package vsphereprivate
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere"
+	"github.com/vmware/govmomi"
+)
+
+// VSphereClient - The VIM/govmomi client.
+type VSphereClient struct {
+	vimClient *govmomi.Client
+}
+
+// ConfigWrapper - wrapping the terraform-provider-vsphere Config struct
+type ConfigWrapper struct {
+	config *vsphere.Config
+}
+
+// NewConfig function
+func NewConfig(d *schema.ResourceData) (*ConfigWrapper, error) {
```

Comment:
I **think** you can simplify this even more:

embed  *vsphere.Config in ConfigWrapper

create an interface:
```go
type VSphereConfig interface{
    vimURL() (*url.URL, error)
    Client() (*VSphereClient, error)
}
```

change `NewConfig` to return the interface `VSphereConfig`

then you don't have to reimplement vimURL() or Client()
(Client looks a little different than the underlying implementation, but I assume it is just a subset and you left out things you didn't need)